Extensions/dnSpy.StringSearcher/StringsControlVM.cs

```diff
@@ -0,0 +1,170 @@
+			public GridViewSortDirection Direction { get; } = Direction;
+
+			public sealed override int Compare(StringReference? x, StringReference? y) {
+				if (x is null && y is null)
+					return 0;
+				if (x is null)
+					return -1;
+				if (y is null)
+					return 1;
+
+				return CompareInternal(x, y);
+			}
+
+			protected virtual int CompareInternal(StringReference x, StringReference y) {
+				int result = Direction switch {
+					GridViewSortDirection.Default or GridViewSortDirection.Ascending => x.Referrer.MDToken.CompareTo(y.Referrer.MDToken),
+					GridViewSortDirection.Descending => y.Referrer.MDToken.CompareTo(x.Referrer.MDToken),
+					_ => throw new ArgumentOutOfRangeException(),
+				};
+
+				if (result != 0) {
+					return result;
+				}
+
+				return Direction switch {
+					GridViewSortDirection.Default or GridViewSortDirection.Ascending => x.Offset.CompareTo(y.Offset),
+					GridViewSortDirection.Descending => y.Offset.CompareTo(x.Offset),
+					_ => throw new ArgumentOutOfRangeException(nameof(Direction)),
+				};
+			}
+		}
+
+		private sealed class ModuleComparer(GridViewSortDirection Direction) : StringReferenceComparer(Direction) {
+			protected override int CompareInternal(StringReference x, StringReference y) {
+				int result = Direction switch {
+					GridViewSortDirection.Ascending => x.Referrer.Module.Name.CompareTo(y.Referrer.Module.Name),
+					GridViewSortDirection.Descending => y.Referrer.Module.Name.CompareTo(x.Referrer.Module.Name),
+					GridViewSortDirection.Default => 0,
+					_ => throw new ArgumentOutOfRangeException(nameof(Direction)),
+				};
+
+				return result == 0
+					? base.CompareInternal(x, y)
+					: result;
+			}
+		}
+
+		private sealed class MethodComparer(GridViewSortDirection Direction) : StringReferenceComparer(Direction) {
+			protected override int CompareInternal(StringReference x, StringReference y) {
+				int result = Direction switch {
+					GridViewSortDirection.Ascending => CompareCore(x, y),
+					GridViewSortDirection.Descending => CompareCore(y, x),
+					GridViewSortDirection.Default => 0,
+					_ => throw new ArgumentOutOfRangeException(nameof(Direction)),
+				};
+
+				return result == 0
+					? base.CompareInternal(x, y)
+					: result;
+			}
+
+			private static int CompareCore(StringReference x, StringReference y) {
+				int result = x.Referrer.DeclaringType.Name.CompareTo(y.Referrer.DeclaringType.Name);
+				if (result == 0)
+					result = x.Referrer.Name.CompareTo(y.Referrer.Name);
+				return result;
+			}
+		}
+
+		private sealed class LiteralComparer(GridViewSortDirection Direction) : StringReferenceComparer(Direction) {
+			protected override int CompareInternal(StringReference x, StringReference y) {
+				int result = Direction switch {
+					GridViewSortDirection.Ascending => x.Literal.CompareTo(y.Literal),
+					GridViewSortDirection.Descending => y.Literal.CompareTo(x.Literal),
```

Comment:
Use `string.Compare(x.Literal, y.Literal, StringComparison.Ordinal)` to force a binary comparison rather than respecting the system culture.


Example:
On .NET Framework:
`"Strasse".CompareTo("Straße")` gives `0` while `string.Compare("Strasse", "Straße", StringComparison.Ordinal)` gives `-108`